sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -278,13 +280,28 @@ class CSVOptions(
     .getOrElse(UNESCAPED_QUOTE_HANDLING, "STOP_AT_DELIMITER").toUpperCase(Locale.ROOT))
 
   /**
+   * Returns true if column pruning is enabled and there are no existence column default values in
+   * the [[schema]].
+   *
    * The column pruning feature can be enabled either via the CSV option `columnPruning` or
    * in non-multiline mode via initialization of CSV options by the SQL config:
    * `spark.sql.csv.parser.columnPruning.enabled`.
    * The feature is disabled in the `multiLine` mode because of the issue:
    * https://github.com/uniVocity/univocity-parsers/issues/529
+   *
+   * We disable column pruning when there are any column defaults, instead preferring to reach in
+   * each row and then post-process it to substitute the default values after.
    */
-  val isColumnPruningEnabled: Boolean = getBool(COLUMN_PRUNING, !multiLine && columnPruning)
+  def isColumnPruningEnabled(schema: StructType): Boolean = {
+    var result = !multiLine && columnPruning
+    if (parameters != null) {
```

Comment:
We shouldn't even try to access to `parameters` when it is `null` because:
1. We never pass `null` as `parameters`.
2. `parameters` are `@transient`, so, it is serialized and apparently not transferred to executors.

Reply:
Good point, I converted this parameter lookup back to a `val` stored in the class and serialized to executors, so we don't try to lookup the CSV options in executors anymore.